source/common/upstream/cluster_manager_impl.cc

```diff
@@ -877,11 +879,12 @@ void ClusterManagerImpl::clusterWarmingToActive(const std::string& cluster_name)
   warming_clusters_.erase(warming_it);
 }
 
-bool ClusterManagerImpl::removeCluster(const std::string& cluster_name) {
+bool ClusterManagerImpl::removeCluster(const std::string& cluster_name, const bool remove_ignored) {
   bool removed = false;
   auto existing_active_cluster = active_clusters_.find(cluster_name);
   if (existing_active_cluster != active_clusters_.end() &&
-      existing_active_cluster->second->added_via_api_) {
+      existing_active_cluster->second->added_via_api_ &&
+      existing_active_cluster->second->ignore_removal_ == remove_ignored) {
```

Comment:
```suggestion
      （!existing_active_cluster->second->ignore_removal_  || remove_ignored)) {
```

does this could be better?

Reply:
This will make cluster to remove when remove_ignored set to true and cluster data has ignore_removal set to false.

This should fine as well. removeCluster will behave as current for cluster with ignore_removal false irrespective of remove_ignored value. And remove_ignored will only take effect for cluster with ignore_removal set to true.